docs/adr/0006-rapid-releases-and-moving-towards-a-monorepo.md

```diff
@@ -0,0 +1,121 @@
+          * Update the version number to be the final release version, e.g. change `v[N].0a1` to  `v[N].0`
+          * Create a GitHub release for that branch
+          * Bump the major version number in main and start a new changelog header.
+    * Phase 2: Automate merges of `application-services` into consumer applications (March 2023 - May 2023)
+      * Every night, for each consumer application (including all mobile apps and Desktop):
+          * Create a PR to take the current application-services code in `main`.
+          * Set up an auto-merge system to automatically merge the PRs.  As with phase 1, no PR should be merged until all PRs pass CI and then all PRs should be merged at once.
+          * Set up an alert that gets sent to `sync-bots` if the test suite fails.  This means we would need to identify why the tests failed and assign a developer to resolve the issue.  The expectation would be that the developer fixes things in time for the next nightly merge.
+          * Note: we currently have automation to do some of this for both iOS and Android.  We should leverage the existing code as much as possible.
+      * Perform the same process whenever a new application-services version is published in a `release_v[N]` branch.
+    * All application-services changes that are breaking for consumers should have corresponding PRs in the consumer repos that have been reviewed and approved.  If we accidentally merge the change into application-services before the PRs are ready, then we should back the change out.
+
+## Decision Outcome
+
+**(B) Move to a rapid release process where changes in app-services are frequently merged into consumer applications**
+
+## Pros and Cons of the Options
+
+### Keep our current release process
+
+  * Good, because it doesn't require changes to our workflow
+  * Bad, for the reasons enumerated at the top of this document
+
+### Move to a rapid release process where changes in app-services are frequently merged into consumer applications
+
+  * Good, because code will get merged and released more frequently.
+  * Good, because we don't need to spend time cutting releases.
+  * Good, because uplifts can be done much quicker.
+  * Good, because more of our release process will be automated.
+  * Bad, because we need to implement new automation.
+  * Bad, because when breaking changes cause the automatic nightly merge to fail for an application.  This means a developer will get the relatively urgent task of resolving things.
+  * Good, because this avoids the current state of affairs where our `main` branch is not ready to be taken by the applications.  This means that a) our current code is not getting used and tested and b) it's very complicated to get other changes to that application, including bugfixes.
+  * Bad, because breaking changes require the corresponding PRs to be approved from all other teams before they can be merged into application-services.
```

Comment:
I know I'm missing something about the process here...but how does this work? Doesn't the breaking change need to be landed and released by application-services before consuming applications can pick up and make their changes?

Reply:
Yeah, I think we'll need to create a way to run the iOS/Android/Desktop CI using an in-progress app-services branch.

For Desktop, I think it would be relatively simple to automate vendoring in our code then running a try build.

For Android/iOS, there's currently code to build/test against an app-services branch, but it needs to be run manually.  the task would be to get it to run in CI for a PR.

I'll add an item in the plan for this.